File: julia/aoc1.jl
#x = [199,200,208,210,200,207,240,269,260,263];




fid1 = open("..\\inputs\\day_1.txt") ; 
x = readlines(fid1) ; 
close(fid1)
x = parse.(Int64,x);
sum(diff(x).> 0) 

using DSP

xc_valid = conv(x,[1;1;1])[3:end-2];
sum(diff(xc_valid).>0)

File: 2015/python/aoc1.py
input_path = '../inputs/day_1.txt'

x = open(input_path).read()

f = 0
ic = 0
for i,c in enumerate(x):
    if c == '(':
        f += 1
    else:
        f -= 1
    if f == -1 and ic == 0:
        ic = i + 1
        
#part1 
print(f)
#part 2   
print(ic)    